sections/sprint-3/goal-position-generation.tex

```diff
@@ -0,0 +1,25 @@
+\section{Goal generation}\label{subsec:goalrefactoring}
+The first implementation of goal generation was done entirely in Unity.
+This means that the goal generation and detection would have been handled on the client-side.
+The idea was to limit the amount of data that needed to be sent across the network.
+But when it was decided that new goal zones should be generated in random locations after a goal had been scored, generating it on the client-side could cause the game state to be different between different players.
```

Comment:
```suggestion
It was then decided that new goal zones should be generated in random locations after a goal had been scored, and by generating it on the client-side it could cause the game state to be different between different players.
```